src/Microsoft.OData.Core/UriParser/Resolver/ODataUriResolver.cs

```diff
@@ -167,7 +164,16 @@ public virtual IEdmTerm ResolveTerm(IEdmModel model, string termName)
                 return term;
             }
 
-            IList<IEdmTerm> results = FindAcrossModels<IEdmTerm>(model, termName, /*caseInsensitive*/ true);
+            IReadOnlyList<IEdmTerm> results;

```

Comment:
this logic is repeated in several places.

Reply:
Which block exactly do you think should be turned into a method, is it the following:
```c#
IReadOnlyList<IEdmTerm> results;
if (model.IsImmutable())
{
    CaseInsensitiveSchemaElementsCache cache = GetCaseInsensitiveSchemaElementsCache(model);
    results = cache.FindTerms(termName);
}
else
{
    results = FindAcrossModels<IEdmTerm>(model, termName, /*caseInsensitive*/ true);
}
```

If that's the one, then the reason I didn't refactor it into a method is that doing so would probably introduce unnecessary complexity and overhead. The main issue is that I'm calling a different cache method depending on the type, (i.e. `cache.FindTerms`, `cache.FindOperations`, `cache.FindSchemaTypes`). So to generalize this block of code to support these different cases I would need to pass a delegate to the method and make sure not to capture the `termName` variable. e.g. 

```c#
void CaseInsensitiveFind<T>(IEdmModel model,  string key, Func<Cache, key, List<T> cacheLookupFunc)  where T : IEdmSchemaElement
{
   if (model.IsImmutable())
   {
       CaseInsensitiveSchemaElementsCache cache = GetCaseInsensitiveSchemaElementsCache(model);
       return cacheLookupFunc(cache, key);
   }

   return FindAcrossModels<T>(model, key, /*caseInsensitive*/ true);
}
```
This is not that complex to be honest but given that this block is fairly simple and straightforward, and used in only 4 places, I didn't feel it was worthwhile to refactor in this way.

But it's not a strong opinion. Given this, do you think it's worth going ahead and refactoring it?

Reply:
If we refactor and add a delegate, make sure it is [static](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/static-anonymous-functions) to not introduce unnecessary delegate or closure allocations on every call.

Reply:
I did a refactor by creating static method and making sure the lambda does not capture the outer scope. Benchmark results remain the same.